### PR TITLE
CDVDgigaherz - check/fix DVD9 functionality (GT4 loading stall)

### PR DESCRIPTION
**Pull request by MrCK1**
~~This bug has been present since at least 2009. The fix was added to the GameDB; but was only applied to Japan game region according to the GameDB. (SCAJ-30007)~~

~~This change updates the GameDB to apply the patch to the US region.~~

Update: Real issue here is that DVD9 do not work on cdvdgigaherz

### CODE REVIEW

**Comment by ssakash** [comment]
why are you modifying the zip file which contains widescreen hacks ? ``bin/cheats_ws.zip``

File: bin/GameIndex.dbf [hunk 1]
```diff
@@ -5180,6 +5180,12 @@ Serial = SCUS-97328
 Name   = Gran Turismo 4
 Region = NTSC-U
 Compat = 5
+//[patches = 77E61C8A]
+//
+//Fixes black screen bug that prevents game from loading
+//patch=1,EE,00100D84,word,24100001 
+//[/patches]
+//
```
Comment (ssakash) [inline bin/GameIndex.dbf]: patch won't be read from the database if you use ``//`` at the beginning of the line. ( currently they're commented out)

**Comment by FlatOutPS2** [comment]
Can you add it to the PAL version as well?
44A61C8F
SCES-51719

File: bin/GameIndex.dbf (hunk 1 above)
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: It's turned off for the Japanese version too.

**Comment by prafullpcsx2** [comment]
Are you sure there is a problem with GT4? It has always worked perfectly fine for me. This is a DVD9 game and videos are on second layer. First time you run PCSX2 it will take about 1 minute to load the second layer so it seems to be stuck on black screen. From second time onwards PCSX2 uses the LayerBreakCache.ini file created in ini folder and there is no waiting.

**Comment by mirh** [comment]
I also recall it was running fine on my PAL copy..

**Comment by FlatOutPS2** [comment]
If it needs 1 minute of waiting the first time, many users will assume it has crashed and give up.
I'm running it from DVD and it definitely doesn't load, but it does with the patch.

**Comment by willkuer** [comment]
Does gigaherz support dvd9? Maybe your game gets only recognized as single layer.

**Comment by prafullpcsx2** [comment]
Oh, now I remember. That's the problem. cdvdgigaherz doesn't support dvd9 correctly.

**Comment by prafullpcsx2** [comment]
Further testing results.

 It only takes about 10 seconds with cdvdpeops (doesn't create ini file), takes about 50 seconds with internal iso reader (creates ini file for further use), it doesn't work with cdvdgigaherz. 

So there is no need to add the patch. Someone just need to take a look at cdvdgigaherz.

**Comment by turtleli** [comment]
Are there any emulog errors that appear?

Potentially CDVDgetDualInfo in CDVDgigaherz returns the wrong information. It sets the layer 1 start address to `*_layer1start = src->GetLayerBreakAddress();` for dual layer discs, but that seems to be the last logical sector on layer 0, not the first logical sector on layer 1. So I think it should be
`*_layer1start = src->GetLayerBreakAddress() + 1;`

(Appveyor build if anyone wants to test - https://ci.appveyor.com/api/buildjobs/p7el2ibxusjp7xvq/artifacts/pcsx2-v1.5.0-dev-426-g021cd8f-cdvd-fixes-212-vs2015-Win32-AppVeyor.7z , I can't test myself since none of my games are dual layer. And this assumes there aren't any other issues.)

**Comment by prafullpcsx2** [comment]
@turtleli : This version doesn't fix it either. I will post the log later.

**Comment by turtleli** [comment]
Ok. thanks for testing.

I suggest someone opens an issue about DVD9 games not working with cdvdgigaherz so we can gather info there, It's unlikely that I'll be able to fix it without a DVD9 game, but I've spotted a few more things which don't seem correct..and maybe that info will help someone else fix it.

**Comment by bositman** [comment]
My suggestion for the 1 minute wait time: https://github.com/PCSX2/pcsx2/issues/799

**Comment by ssakash** [comment]
@MrCK1 

> ~~This bug has been present since at least 2009. The fix was added to the GameDB; but was only applied to Japan game region according to the GameDB. (SCAJ-30007)
This change updates the GameDB to apply the patch to the US region.~~
Update: Real issue here is that DVD9 do not work on cdvdgigaherz

I personally think opening an issue would be better in that case as turtleli suggested earlier. Pull requests are only for merging patches to the master branch **not** for reporting issues.

**Comment by MrCK1** [comment]
Yeah I'll just close it for now. If someone is interested, they can open an issue

**Pull request closed without merging**